### ISSUE TITLE
Support disabling polling of materials of pipelines created via SCM Plugin

### ISSUE THREAD

**Issue by peterrosell** [gocd/gocd#11635]
##### Issue Type

Bug report/Feature enhancement

##### Summary

I want to disable polling of materials of pipelines that are configured via a SCM Plugin and instead rely on web-hooks.

It works for manually created pipelines it's possible to set `Repository polling behavior` to `Fetch updates to this repository only on webhook or manual trigger`, but not when creating pipelines via a SCM plugin like [gocd-yaml-config-plugin](https://github.com/tomzo/gocd-yaml-config-plugin) or [gocd-json-config-plugin](https://github.com/tomzo/gocd-json-config-plugin). When setting the `auto_update` flag in the config file it's silently ignored and always set to true in the internal data model.

##### Environment

GoCD Version: 23.2.0
YAML Configuration Plugin: 0.14.1
JSON Configuration Plugin: 0.6.0

##### Steps to Reproduce

1. Create a Pipeline in a repo and add it as a config repo.
```
format_version: 9
pipelines:
  pipe-test-1:
    group: TestGroup
    lock_behaviour: none

    materials:
      pipe-test-pr:
        plugin_configuration:
          id: github.pr
          version: 1
        options:
          url: git@github.com:peterrosell/test-repo.git
        auto_update: false
    stages:
      - test-pipe2:
          clean_workspace: true
          jobs:
            release:
              tasks:
                - exec:
                    command: /bin/bash
                    arguments:
                      - -c
                      - ls
```
2. Fetch the SCM config with `curl` and see that `auto_update` is set to `true`.
```
curl -s 'http://localhost:8153/go/api/admin/scms' -H 'Accept:application/vnd.go.cd.v4+json' | jq '.'
```

##### Expected Results

The `auto_update` flag should be set to `false`.

##### Actual Results

The `auto_update` flag is set to `true`.

##### Possible Fix

I've been digging around in the code and the flag doesn't exists in [`CRPluggableScmMaterial`](https://github.com/gocd/gocd/blob/077031dd54e27fba8c6e99d4dcc0275b0e5f9561/plugin-infra/go-plugin-config-repo/src/main/java/com/thoughtworks/go/plugin/configrepo/contract/material/CRPluggableScmMaterial.java#L34) like it does in [`CRScmMaterial`](https://github.com/gocd/gocd/blob/077031dd54e27fba8c6e99d4dcc0275b0e5f9561/plugin-infra/go-plugin-config-repo/src/main/java/com/thoughtworks/go/plugin/configrepo/contract/material/CRScmMaterial.java#L43).
So by adding this property to `CRPluggableScmMaterial` and then make sure the flag propagates from `PluggableSCMMaterialConfig` to the `CRPluggableScmMaterial`.

Affected code in `ConfigConverter.java` [here](https://github.com/gocd/gocd/blob/077031dd54e27fba8c6e99d4dcc0275b0e5f9561/server/src/main/java/com/thoughtworks/go/config/ConfigConverter.java#L975) and [here](https://github.com/gocd/gocd/blob/077031dd54e27fba8c6e99d4dcc0275b0e5f9561/server/src/main/java/com/thoughtworks/go/config/ConfigConverter.java#L398).

I have done some tests locally and got it to work so will create a PR to fix this issue.

**Comment by MPV** [comment]
@chadlwilson Thanks for investigating this. 
What do you think is the next step forward here?

**Comment by chadlwilson** [comment]
See the discussion at https://github.com/gocd/gocd/pull/11636#issuecomment-1586898326 and below in particular. The extent of my thoughts are on the PR, and haven't got any further than that.

There's probably not really a next step without addressing the concerns on approach, which I'm not sure how to do right now (or put differently, have not had the time/energy/inclination to figure out).

The benefit appears somewhat small (given how long the feature has been around and no-one highlighting until now, that I can find!), and at the moment the effort/risk is relatively high, so we need to find a way to change that equation, which is rather difficult with the resources available.

TL;DR The current proposed approach in the PR has at least two problems
- can silently mutate configuration outside the config repository's scope, including materials that are not within scope of a given change, unintentionally affecting other pipelines.
- _appears to_ mutate shared configuration objects in memory prior to a change being fully validated as safe to merge into the configuration, in a way that is not reverted/undone if the change later fails a validation constraint (this requires specific validation)

That's not really a solution I am comfortable with given the chance for unintended consequences that will be difficult for a user to reason about or debug.
